storage/indexer/src/event_v2_translator.rs

```diff
@@ -0,0 +1,1327 @@
+            let sequence_number = engine
+                .get_next_sequence_number(&key, coin_store_resource.withdraw_events().count())?;
+            (key, sequence_number)
+        } else {
+            (EventKey::new(3, *coin_withdraw.account()), 0)
+        };
+        let withdraw_event = WithdrawEvent::new(coin_withdraw.amount());
+        Ok(ContractEventV1::new(
+            key,
+            sequence_number,
+            WITHDRAW_EVENT_TYPE.clone(),
+            bcs::to_bytes(&withdraw_event)?,
+        ))
+    }
+}
+
+struct CoinRegisterTranslator;
+impl EventV2Translator for CoinRegisterTranslator {
+    fn translate_event_v2_to_v1(
+        &self,
+        v2: &ContractEventV2,
+        engine: &EventV2TranslationEngine,
+    ) -> Result<ContractEventV1> {
+        let coin_register = CoinRegister::try_from_bytes(v2.event_data())?;
+        let struct_tag_str = "0x1::account::Account".to_string();
+        let struct_tag = StructTag::from_str(&struct_tag_str)?;
+        let (key, sequence_number) = if let Some(state_value_bytes) =
+            engine.get_state_value_bytes_for_resource(coin_register.account(), &struct_tag)?
+        {
+            let account_resource: AccountResource = bcs::from_bytes(&state_value_bytes)?;
+            let key = *account_resource.coin_register_events().key();
+            let sequence_number = engine
+                .get_next_sequence_number(&key, account_resource.coin_register_events().count())?;
+            (key, sequence_number)
+        } else {
+            (EventKey::new(0, *coin_register.account()), 0)
+        };
+        let coin_register_event = CoinRegisterEvent::new(coin_register.type_info().clone());
+        Ok(ContractEventV1::new(
+            key,
+            sequence_number,
+            COIN_REGISTER_EVENT_TYPE.clone(),
+            bcs::to_bytes(&coin_register_event)?,
+        ))
+    }
+}
+
+struct KeyRotationTranslator;
+impl EventV2Translator for KeyRotationTranslator {
+    fn translate_event_v2_to_v1(
+        &self,
+        v2: &ContractEventV2,
+        engine: &EventV2TranslationEngine,
+    ) -> Result<ContractEventV1> {
+        let key_rotation = KeyRotation::try_from_bytes(v2.event_data())?;
+        let struct_tag_str = "0x1::account::Account".to_string();
+        let struct_tag = StructTag::from_str(&struct_tag_str)?;
+        let (key, sequence_number) = if let Some(state_value_bytes) =
+            engine.get_state_value_bytes_for_resource(key_rotation.account(), &struct_tag)?
+        {
+            let account_resource: AccountResource = bcs::from_bytes(&state_value_bytes)?;
+            let key = *account_resource.key_rotation_events().key();
+            let sequence_number = engine
+                .get_next_sequence_number(&key, account_resource.key_rotation_events().count())?;
+            (key, sequence_number)
+        } else {
+            (EventKey::new(1, *key_rotation.account()), 0)
```

Comment:
ditto

Reply:
It's the creation number of the event key for `KeyRotationEvent`. It's deterministically 1.